src/Panel/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx

```diff
@@ -0,0 +1,496 @@
+import * as React from 'react';
+
+import moment from 'moment';
+
+const _isFinite = require('lodash/isFinite');
+const _isEqual = require('lodash/isEqual');
+
+import {
+  TimeSlider,
+  timeLayerAware,
+  ToggleButton,
+  SimpleButton
+} from '../../index';
+
+import {
+  Select,
+  DatePicker,
+  Popover
+} from 'antd';
+const RangePicker = DatePicker.RangePicker;
+const Option = Select.Option;
+
+import './TimeLayerSliderPanel.less';
+
+type timeRange = [moment.Moment, moment.Moment];
+
+export interface Tooltips {
+  hours: string;
+  days: string;
+  weeks: string;
+  months: string;
+  years: string;
+  setToNow: string;
+  dataRange: string;
+}
+
+export interface DefaultTimeLayerSliderPanelProps {
+  className: string;
+  onChange: (arg: moment.Moment) => void;
+  timeAwareLayers: any[];
+  value: moment.Moment;
+  dateFormat: string;
+  tooltips: Tooltips;
+}
+
+export interface TimeLayerSliderPanelProps extends Partial<DefaultTimeLayerSliderPanelProps> {
+  map: any;
+  initStartDate: moment.Moment;
+  initEndDate: moment.Moment;
+}
+
+export interface TimeLayerSliderPanelState {
+  value: moment.Moment;
+  playbackSpeed: string;
+  autoPlayActive: boolean;
+  startDate: moment.Moment;
+  endDate: moment.Moment;
+}
+
+/**
+ * mapStateToProps - mapping state to props of Map Component.
+ *
+ * @param {Object} state current state
+ * @return {Object} mapped props
+ */
+
+/**
+ * The panel combining all time slider related parts.
+ */
+export class TimeLayerSliderPanel extends React.Component<TimeLayerSliderPanelProps, TimeLayerSliderPanelState> {
+
+  private _TimeLayerAwareSlider: any;
+  private _wmsTimeLayers: any[];
+  private _interval: number;
+
+  /**
+   * The default props of LayerSetBaseMapChooser
+   *
+   * @static
+   * @type {DefaultLayerSetBaseMapChooserProps}
+   * @memberof LayerSetBaseMapChooser
+   */
+  public static defaultProps: DefaultTimeLayerSliderPanelProps = {
+    className: '',
+    onChange: () => {/* empty */},
```

Comment:
```suggestion
    onChange: () => {},
```

Reply:
I think it should be `onChange: (value: moment.Moment) => void,`

Reply:
If it is empty it shouldn't be a `defaultProp`

Reply:
If it is set like suggested, the linting fails:` no-empty: block is empty`. The proposed solutions was to enter an comment into this block.
If these are removed completely, compilation fails, because this information is missing in `TimeLayerAware`. Thus, I would suggest to leave them as they are, or do you have another suggestion @KaiVolland ?

Reply:
Set the `tslint.json` to allow empty functions.